app/controllers/clubs_controller.rb

```
@@ -0,0 +1,103 @@
+class ClubsController < ApplicationController
+  # GET /clubs
+  # GET /clubs.json
+  def index
+    @clubs = Club.all
+  end
+
+  # GET /clubs/1
+  # GET /clubs/1.json
+  def show
+    @club = Club.find(params[:id])
+  end
+
+  # GET /clubs/new
+  def new
+    @club = Club.new
+  end
+
+  # GET /clubs/1/edit
+  def edit
+    @club = Club.find(params[:id])
+  end
+
+  # POST /clubs
+  # POST /clubs.json
+  def create
+    @club = Club.new(club_params)
+
+    respond_to do |format|
+      if @club.save
+        format.html { redirect_to @club, notice: 'Club was successfully created.' }
+        format.json { render :show, status: :created, location: @club }
+      else
+        format.html { render :new }
+        format.json { render json: @club.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /clubs/1
+  # PATCH/PUT /clubs/1.json
+  def update
+    @club = Club.find(params[:id])
+    respond_to do |format|
+      if @club.update(club_params)
+        format.html { redirect_to @club, notice: 'Club was successfully updated.' }
+        format.json { render :show, status: :ok, location: @club }
+      else
+        format.html { render :edit }
+        format.json { render json: @club.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /clubs/1
+  # DELETE /clubs/1.json
+  def destroy
+    @club = Club.find(params[:id])
+    @club.destroy
+    respond_to do |format|
+      format.html { redirect_to clubs_url, notice: 'Club was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  # POST /clubs/1/join
+  # POST /clubs/1/join.json
+  def join
+    @club = Club.find(params[:id])
+    ClubMembership.create({
+      user_id: current_user.id,
+      club_id: @club.id
+    })
```

I think you can do this simpler, something like:
```ruby
@club.memberships.create! current_user
```
That might not work, you might have to:
```ruby
@club.memberships.create! user_id: current_user.id
```

---

As an aside, the method `create!` with an exclamation will throw an error if the save fails, at the moment yours will fail silently.